packages/core/admin/admin/src/utils/createRoute.tsx

```diff
@@ -16,10 +21,7 @@ const LazyCompo = ({ loadComponent }) => {
         // We will move the entire codebase to use named exports only
         // Until then we support both cases with priority given to the existing default exports
         setCompo(() => loadedCompo?.default ?? loadedCompo);
```

Comment:
@joshuaellis I need your help here, because this will create an error: do you have an idea how to type this? 

```Argument of type '() => React.ElementType<any> | { default?: React.ElementType<any> | undefined; }' is not assignable to parameter of type 'SetStateAction<ElementType<any> | null>'.
  Type '() => React.ElementType<any> | { default?: React.ElementType<any> | undefined; }' is not assignable to type 'FunctionComponent<any>'.
    Type 'ElementType<any> | { default?: ElementType<any> | undefined; }' is not assignable to type 'ReactElement<any, any> | null'.
      Type 'string' is not assignable to type 'ReactElement<any, any>'.
```

Same for the case where the Component is called:

```jsx
<Route
      render={() => <LazyCompo loadComponent={Component} />}
      key={to}
      path={to}
      exact={exact || false}
    />
```

errors at the prop:

```
Type 'ElementType<any>' is not assignable to type '() => Promise<{ default?: ElementType<any> | undefined; }>'.
  Type 'string' is not assignable to type '() => Promise<{ default?: ElementType<any> | undefined; }>'.ts(2322)
createRoute.tsx(11, 3): The expected type comes from property 'loadComponent' which is declared here on type 'IntrinsicAttributes & { loadComponent: () => Promise<{ default?: ElementType<any> | undefined; }>; }'
```

Reply:
iirc ElementType can be `"div"`, so i've gone for `ComponentType` instead...